core/domain/draft_upgrade_services.py

```diff
@@ -225,37 +243,43 @@ def _convert_html_in_draft_change_list(
                 # This code below should be updated if any new interaction
                 # is allowed to have HTML in the solution correct answer
                 # The typecheckings below can be avoided once #9413 is fixed.
-                if new_value['correct_answer']:
-                    if isinstance(new_value['correct_answer'], list):
+                if solution['correct_answer']:
+                    if isinstance(solution['correct_answer'], list):
                         for list_index, html_list in enumerate(
-                                new_value['correct_answer']):
+                                solution['correct_answer']):
                             if isinstance(html_list, list):
                                 for answer_html_index, answer_html in enumerate(
                                         html_list):
                                     if isinstance(answer_html, str):
-                                        new_value['correct_answer'][list_index][
+                                        # Here we use MyPy ignore because
```

Comment:
To avoid this, every time you use isinstance, cast to a narrower union type (make a new subtype for answers that are lists, etc.).

Conceptually, that is what should be happening with isinstance, so we do it manually instead. But that is better logically than using `type: ignore`.

Reply:
Instead of using two casts, I defined one cast at the top of the first `isinstance` with reasoning.